### PR TITLE
Document remaining node queries.

### LINKED ISSUE #2228: Document remaining C Tree node queries
Following up on #2180. Part of #2140

### PR DESCRIPTION
**Pull request by jeromekelleher**
Closes #2228

### CODE REVIEW

File: c/tskit/trees.h [hunk 1]
```diff
@@ -764,15 +764,99 @@ void tsk_tree_print_state(const tsk_tree_t *self, FILE *out);
 @{
 */
 
-/* Returns true if u is a descendant of v; false otherwise */
-bool tsk_tree_is_descendant(const tsk_tree_t *self, tsk_id_t u, tsk_id_t v);
-bool tsk_tree_is_sample(const tsk_tree_t *self, tsk_id_t u);
+/**
+@brief Returns the parent of the specified node.
```
Comment (benjeffery) [inline c/tskit/trees.h:768]: The return value is the err code.
```suggestion
@brief Gets the parent of the specified node.
```

File: c/tskit/trees.h [hunk 2]
```diff
@@ -764,15 +764,99 @@ void tsk_tree_print_state(const tsk_tree_t *self, FILE *out);
...
+
+@param self A pointer to a tsk_tree_t object.
+@param u The tree node.
+@param parent A tsk_id_t pointer to store the returned parent node.
+@return 0 on success or a negative value on failure.
+*/
+int tsk_tree_get_parent(const tsk_tree_t *self, tsk_id_t u, tsk_id_t *parent);
+
+/**
+@brief Returns the time of the specified node.
```
Comment (benjeffery) [inline c/tskit/trees.h:784]:
```suggestion
@brief Gets the time of the specified node.
```

File: c/tskit/trees.h [hunk 3]
```diff
@@ -764,15 +764,99 @@ void tsk_tree_print_state(const tsk_tree_t *self, FILE *out);
...
 
+@param self A pointer to a tsk_tree_t object.
+@param u The tree node.
+@param ret_time A double pointer to store the returned node time.
+@return 0 on success or a negative value on failure.
+*/
+int tsk_tree_get_time(const tsk_tree_t *self, tsk_id_t u, double *ret_time);
+
+/**
+@brief Return number of nodes on the path from the specified node to root.
```
Comment (benjeffery) [inline c/tskit/trees.h:824]:
```suggestion
@brief Compute the number of nodes on the path from the specified node to root.
```

File: c/tskit/trees.h [hunk 4]
```diff
@@ -764,15 +764,99 @@ void tsk_tree_print_state(const tsk_tree_t *self, FILE *out);
...
+
+@param self A pointer to a tsk_tree_t object.
+@param u The tree node.
+@param ret_depth An int pointer to store the returned node depth.
+@return 0 on success or a negative value on failure.
+*/
+int tsk_tree_get_depth(const tsk_tree_t *self, tsk_id_t u, int *ret_depth);
+
+/**
+@brief Return the length of the branch ancestral to the specified node.
```
Comment (benjeffery) [inline c/tskit/trees.h:842]:
```suggestion
@brief Compute the length of the branch ancestral to the specified node.
```

File: c/tskit/trees.h [hunk 5]
```diff
@@ -764,15 +764,99 @@ void tsk_tree_print_state(const tsk_tree_t *self, FILE *out);
...
+@param ret_time A double pointer to store the returned node time.
+@return 0 on success or a negative value on failure.
+*/
+int tsk_tree_get_time(const tsk_tree_t *self, tsk_id_t u, double *ret_time);
+
+/**
+@brief Return number of nodes on the path from the specified node to root.
+
+@rst
+Return the number of nodes on the path from u to root, not including u.
```
Comment (benjeffery) [inline c/tskit/trees.h:827]:
```suggestion
Compute the number of nodes on the path from u to root, not including u.
```

File: c/tskit/trees.h [hunk 6]
```diff
@@ -764,15 +764,99 @@ void tsk_tree_print_state(const tsk_tree_t *self, FILE *out);
...
+@param ret_depth An int pointer to store the returned node depth.
+@return 0 on success or a negative value on failure.
+*/
+int tsk_tree_get_depth(const tsk_tree_t *self, tsk_id_t u, int *ret_depth);
+
+/**
+@brief Return the length of the branch ancestral to the specified node.
+
+@rst
+Return the length of the branch ancestral to the specified node.
```
Comment (benjeffery) [inline c/tskit/trees.h:845]:
```suggestion
Compute the length of the branch ancestral to the specified node.
```

File: c/tskit/trees.h [hunk 7]
```diff
@@ -788,20 +872,65 @@ leaf node is zero.
...
 
-int tsk_tree_get_time(const tsk_tree_t *self, tsk_id_t u, double *t);
-int tsk_tree_get_parent(const tsk_tree_t *self, tsk_id_t u, tsk_id_t *parent);
-int tsk_tree_get_depth(const tsk_tree_t *self, tsk_id_t u, int *depth);
-int tsk_tree_get_mrca(const tsk_tree_t *self, tsk_id_t u, tsk_id_t v, tsk_id_t *mrca);
+/**
+@brief Counts the number of samples in the subtree rooted at a node.
+
+@rst
+Returns the number of samples descending from a particular node,
```
Comment (benjeffery) [inline c/tskit/trees.h:887]:
```suggestion
Counts the number of samples descending from a particular node,
```

File: c/tskit/trees.h [hunk 8]
```diff
@@ -788,20 +872,65 @@ leaf node is zero.
...
 */
 int tsk_tree_get_total_branch_length(
     const tsk_tree_t *self, tsk_id_t u, double *ret_tbl);
 
-int tsk_tree_get_time(const tsk_tree_t *self, tsk_id_t u, double *t);
-int tsk_tree_get_parent(const tsk_tree_t *self, tsk_id_t u, tsk_id_t *parent);
-int tsk_tree_get_depth(const tsk_tree_t *self, tsk_id_t u, int *depth);
-int tsk_tree_get_mrca(const tsk_tree_t *self, tsk_id_t u, tsk_id_t v, tsk_id_t *mrca);
+/**
+@brief Counts the number of samples in the subtree rooted at a node.
```
Comment (benjeffery) [inline c/tskit/trees.h:884]:
```suggestion
@brief Count the number of samples in the subtree rooted at a node.
```

**Review by benjeffery — APPROVED** [review]
LGTM, just a few places where we aren't returning the value but the docs say we are.

**Comment by jeromekelleher** [comment]
Done, thanks.

**Pull request merged**